Examples/Python/src/Geometry.cpp

```diff
@@ -146,6 +146,13 @@ void addGeometry(Context& ctx) {
   {
     py::class_<Acts::TrackingGeometry, std::shared_ptr<Acts::TrackingGeometry>>(
         m, "TrackingGeometry")
+        .def(py::init([](const MutableTrackingVolumePtr& v,
+                         // std::shared_ptr<const IMaterialDecorator> m,
+                         const GeometryIdentifierHook& h,
+                         const PythonLogger& logger) {
+          return std::make_shared<Acts::TrackingGeometry>(v, nullptr, h,
+                                                          logger.logger());
+        }))
```

Comment:
Don't use `PythonLogger` as argument here. Also, can't you just bind the constructor without a separate functions? Also also, I don't think we should be creating these objects from python.

Reply:
The constructor takes a reference to a logger. That means, the logger must live on the call-site. This is most naturally achieved by our logger that lives in python.
Option B would be to let the callee take a `std::unique_ptr<Logger>`, but that would require technically an API break I think...

Reply:
I think what we usually do is pass in a log level and construct the logger in the binding. The logger will not be hold on in C++ so that should be safe